x/cronos/keeper/grpc_query.go

```diff
@@ -1,7 +1,39 @@
 package keeper
 
 import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/crypto-org-chain/cronos/x/cronos/types"
+	"github.com/ethereum/go-ethereum/common"
 )
 
 var _ types.QueryServer = Keeper{}
+
+// ContractByDenom query contract by denom, returns both external contract and auto deployed contract
+func (k Keeper) ContractByDenom(goCtx context.Context, req *types.ContractByDenomRequest) (*types.ContractByDenomResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	rsp := types.ContractByDenomResponse{}
+	contract, found := k.getExternalContractByDenom(ctx, req.Denom)
+	if found {
+		rsp.Contract = contract.String()
+	}
+	autoContract, found := k.getAutoContractByDenom(ctx, req.Denom)
+	if found {
+		rsp.AutoContract = autoContract.String()
+	}
+	return &rsp, nil
```

Comment:
could return an error in case it can not find in both map to be consistent with `DenomByContract`

Reply:
done